Library/Homebrew/utils/analytics.sh

```diff
@@ -0,0 +1,67 @@
+setup-analytics() {
+  if [[ -z "$HOMEBREW_ANALYTICS" || -n "$HOMEBREW_NO_ANALYTICS" ]]
+  then
+    return
+  fi
+
+  # User UUID file. Used for Homebrew user counting. Can be deleted and
+  # recreated with no adverse effect (beyond our user counts being inflated).
+  HOMEBREW_ANALYTICS_USER_UUID_FILE="$HOME/.homebrew_analytics_user_uuid"
+  if [[ -r "$HOMEBREW_ANALYTICS_USER_UUID_FILE" ]]
+  then
+    HOMEBREW_ANALYTICS_USER_UUID="$(<"$HOMEBREW_ANALYTICS_USER_UUID_FILE")"
+  else
+    HOMEBREW_ANALYTICS_USER_UUID="$(uuidgen)"
+    echo "$HOMEBREW_ANALYTICS_USER_UUID" > "$HOMEBREW_ANALYTICS_USER_UUID_FILE"
+  fi
+  export HOMEBREW_ANALYTICS_ID="UA-75654628-1"
+  export HOMEBREW_ANALYTICS_USER_UUID
+}
+
+report-analytics-screenview-command() {
+  if [[ -z "$HOMEBREW_ANALYTICS" || -n "$HOMEBREW_NO_ANALYTICS" ]]
+  then
+    return
+  fi
+
+  # Don't report non-official commands.
+  if ! [[ "$HOMEBREW_COMMAND" = "bundle"   ||
+          "$HOMEBREW_COMMAND" = "cask"     ||
+          "$HOMEBREW_COMMAND" = "services" ||
+          -f "$HOMEBREW_LIBRARY/Homebrew/cmd/$HOMEBREW_COMMAND.rb"     ||
+          -f "$HOMEBREW_LIBRARY/Homebrew/cmd/$HOMEBREW_COMMAND.sh"     ||
+          -f "$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOMEBREW_COMMAND.rb" ||
+          -f "$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOMEBREW_COMMAND.sh" ]]
+  then
+    return
+  fi
+
+  local ARGS=(
+    --max-time 3 \
+    --user-agent "$HOMEBREW_USER_AGENT_CURL" \
+    -d v="1" \
+    -d tid="$HOMEBREW_ANALYTICS_ID" \
+    -d cid="$HOMEBREW_ANALYTICS_USER_UUID" \
+    -d aip="1" \
+    -d an="$HOMEBREW_PRODUCT" \
+    -d av="$HOMEBREW_VERSION" \
+    -d t="screenview" \
```

Comment:
Nitpick, but quoting a literal string without white space or other special characters feels weird to me. Same for the other two instances above. (I guess I can see an argument for consistency with the other `-d` arguments, though.)